### ISSUE TITLE
CVE-2016-7056

### ISSUE THREAD

**Issue by MSe1969** [derfelot/discussion#15]
CVE-2016-7056 (from May 2017 ASB) points to boringssl. 
It was left out (N/A for LP), but in fact - see #3 - it is relevant. 
In LOS 13, we have deferred it, since a backport seemed to be too complicated and most probably not relevant. However, Google has published tag android-6.0.1_r80 recently, which makes the 6.0.1 coding fix for this CVE visible and we can now analyze and evaluate the LP backporting again.

**Comment by Micha-Btz** [comment]
maybe we should test if we can use the library directly from the cm-13.0 repo. Would save us some headache.

**Comment by MSe1969** [comment]
'cm-13.0' work on tag android-6.0.1_r80 is still WIP in LineageOS.
between Android 5.x and 6.x, the most significant visible difference of the boringssl lib is its location in the build tree (and this is my suspicion, why Google may not have considered LP). But yes, we can try...

**Comment by Micha-Btz** [comment]
I have git the LOS boringssl cm-13.0 a try. build and install fine. But wifi is not working. Mobile connection with LTE works fine. Log tells that the wifi state machine is unclear maybe a chipper problem with WPA2.
Seems not to work for us.

**Comment by MSe1969** [comment]
ok, thanks for testing. When I find some time, I'll try to backport.

**Comment by Micha-Btz** [comment]
Where you able to test this @MSe1969 ?

**Comment by MSe1969** [comment]
Not yet, unfortunately - I have now a little bit more time, so will try to work on it this week.

**Comment by MSe1969** [comment]
I have applied (and backported) the boringssl changes of tag android-6.0.1_r80 into the cm-12.1-test branch of the *derfelot/android_external_chromium_org_third_party_boringssl_src* repo.
Created a test build, which I'll flash now.

**Comment by MSe1969** [comment]
Test build seems to run fine - tested web browsers (Firefox, PrivacyBrowser) https and logon via https, all looks good, so we could ship with the September ASB.

The impact in general however is low or even neglectible:
Up to Android 5, OpenSSL was used as the general SSL solution and Boringssl was only used for Chromium components (which we anyhow now pull as **pre-built webview.apk** from CM-14.1). Only starting with Android 6, Boringssl fully replaces OpenSSL, which also explains, why the repository was moved from external/chromium_org/third_party/boringssl/rsc to external/boringssl.

Nevertheless, patch has been applied.

**Comment by Micha-Btz** [comment]
My test build run fine too. So I think we can ship this with ASB 09/2017.

**Comment by MSe1969** [comment]
My suspicion became confirmed: 
The code is not even used any longer, since we have specified to use the pre-built webview.apk - I had the impression already when looking at the make files. So I made a test and simply entered the following statement into crypto/ec/ec.c: `#include "Ei gude, wie?"` - this should normally abort the build with a syntax error during compile, but the build went through without any issue.
So I'll delete the cm-12.1-test branch from this repo and we can also remove the repo from our local manifest.
